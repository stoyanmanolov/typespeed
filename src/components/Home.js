import React from "react";
import { useHistory } from "react-router-dom";

const Home = () => {
  let history = useHistory();
  const handleClick = event => {
    history.push("/race");
  };

  return (
    <div className="container">
      <h2>Test your typing speed</h2>
      <p>
        This app determines your speed of typing measured in WPM (words per
        minute). <br /> Inspired by <i>typeracer.com</i>
      </p>
      <button className="start-button" onClick={handleClick}>
        START THE RACE
      </button>
    </div>
  );
};

export default Home;
