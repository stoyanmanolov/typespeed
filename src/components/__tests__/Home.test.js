import React from "react";
import { shallow } from "enzyme";
import Home from "../Home";

let mockedPush = jest.fn();
jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: mockedPush
  })
}));

describe("The Home component", () => {
  let wrapper = shallow(<Home />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe("when .start-button is clicked", () => {
    it("redirects to the '/race' page", () => {
      wrapper.find(".start-button").simulate("click");
      expect(mockedPush).toHaveBeenCalledWith("/race");
    });
  });
});
