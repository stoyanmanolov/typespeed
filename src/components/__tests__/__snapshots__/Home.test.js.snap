// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home component renders correctly 1`] = `ShallowWrapper {}`;
